common/src/main/java/io/seata/common/loader/EnhancedServiceLoader.java

```diff
@@ -198,181 +173,446 @@
+         */
+        private List<S> loadAll(Class[] argsType, Object[] args) {
+            List<S> allInstances = new ArrayList<>();
+            List<Class> allClazzs = getAllExtensionClass();
+            if (CollectionUtils.isEmpty(allClazzs)) {
+                return allInstances;
+            }
+            try {
+                for (Class clazz : allClazzs) {
+                    ExtensionDefinition definition = classToDefinitionMap.get(clazz);
+                    allInstances.add(getExtensionInstance(definition, findClassLoader(), argsType, args));
                 }
+            } catch (Throwable t) {
+                throw new EnhancedServiceNotFoundException(t);
+            }
+            return allInstances;
+        }
+
+        /**
+         * Get all the extension classes, follow {@linkplain LoadLevel} defined and sort order
+         *
+         * @return all extension class
+         */
+        @SuppressWarnings("rawtypes")
+        private List<Class> getAllExtensionClass() {
+            return loadAllExtensionClass(findClassLoader());
+        }
+
+        /**
+         * Get all the extension classes, follow {@linkplain LoadLevel} defined and sort order
+         *
+         * @param loader the loader
+         * @return all extension class
+         */
+        @SuppressWarnings("rawtypes")
+        private List<Class> getAllExtensionClass(ClassLoader loader) {
+            return loadAllExtensionClass(loader);
+        }
 
-                extensions = activateExtensions;
+        @SuppressWarnings("rawtypes")
+        private S loadExtension(ClassLoader loader, Class[] argTypes,
+                                Object[] args) {
+            try {
+                loadAllExtensionClass(loader);
+                ExtensionDefinition defaultExtensionDefinition = getDefaultExtensionDefinition();
+                return getExtensionInstance(defaultExtensionDefinition, loader, argTypes, args);
+            } catch (Throwable e) {
+                if (e instanceof EnhancedServiceNotFoundException) {
+                    throw (EnhancedServiceNotFoundException)e;
+                } else {
+                    throw new EnhancedServiceNotFoundException(
+                            "not found service provider for : " + type.getName() + " caused by " + ExceptionUtils
+                                    .getFullStackTrace(e));
+                }
             }
+        }
 
-            if (extensions.isEmpty()) {
-                throw new EnhancedServiceNotFoundException(
-                    "not found service provider for : " + service.getName() + "[" + activateName
-                        + "] and classloader : " + ObjectUtils.toString(loader));
+        @SuppressWarnings("rawtypes")
+        private S loadExtension(String activateName, ClassLoader loader, Class[] argTypes,
+                                Object[] args) {
+            if (io.seata.common.util.StringUtils.isEmpty(activateName)) {
```

Comment:
In the original logic, this could be empty.

Reply:
This method is used for scenarios where the user explicitly passes activateName, not the initial loaded scenario.